Highlighter: Move writing of top-level highlight field to SearchSourceBuilder

cbuescher opened the pull request:
Most elements in SearchSourceBuilder (e.g. aggs, queries) write their top-level ParseField name in toXContent(), while HighlightBuilder used to do it in its own toXContent() method. Moved this up so SeachSourceBuilder for consistency.

cbuescher commented:
Stumbled upon this while going through the SearchSourceBuilder, I'd be interested to learn if there are any reasons for writing the top-level field name in the Highlighter itself so far, so just opening this PR as a question.

nik9000 commented:
I don't know why it did that. Those reasons might be lost to the depths of time. I say if `git blame` doesn't give you any hints on another person to ask then just merge this to make it consistent.

cbuescher commented:
@nik9000 I was just wondering if highlighter#toXContent was used anywhere else besides the SearchSourceBuilder, couldn't find anything really.

nik9000 commented:
LGTM

cbuescher committed:
Highlighter: Move xContent write of highlight fieldname to SearchSourceBuilder

Most elements in SearchSourceBuilder (e.g. aggs, queries) write their top-level
ParseField name in toXContent(), while HighlightBuilder used to do it in
its own toXContent() method. Moved this up so SeachSourceBuilder for consistency.
